registry/consul/service.go

```diff
@@ -39,7 +39,10 @@ func servicesConfig(client *api.Client, checks []*api.HealthCheck, tagPrefix str
 	// map service name to list of service passing for which the health check is ok
 	m := map[string]map[string]bool{}
 	for _, check := range checks {
-		name, id := check.ServiceName, check.ServiceID
+		// Make the node part of the id, because according to the Consul docs
+		// the ServiceID is unique per agent but not cluster wide
+		// https://www.consul.io/api/agent/service.html#id
+		name, id := check.ServiceName, check.ServiceID+check.Node
```

Comment:
If you want to align with what Consul servers internally do:
`fmt.Sprintf("%s.%s", check.ServiceID, check.Node)`

Reply:
I'll fix that in a subsequent commit.